Replicator/c4Replicator.hh

```diff
@@ -97,7 +134,15 @@ struct C4Replicator : public RefCounted,
 
     virtual void stop() {
         LOCK(_mutex);
+        _cancelStop = false;
+        if(_status.level == kC4Stopping) {
+            // Already stopping, this call is spammy so ignore it
+            logVerbose("Duplicate call to stop() ignored...");
+            return;
+        }
+        
         if (_replicator) {
+            _status.level = kC4Stopping;
             _replicator->stop();
         } else if (_status.level != kC4Stopped) {
```

Comment:
That seems unnecessarily complex?  Just set it to kC4Stopped.  Nobody cares what it was, before.

Reply:
Can't, it's not all the way stopped yet.  This next call to stop is async.

Reply:
No, not what I meant. `else if (_status.level != kC4Stopped) {`
sets the status to kC4Stopped unless the status is kC4Stopped.  Just set it to kC4Stopped.  I.e:

else `{ _status.level = kC4Stopped }`

Reply:
I'd like to keep the review within the scope of what I changed, but that would result in spurious notifications back to the delegate (inside that block calls `notifyStateChanged`)

Reply:
Gotcha on both accounts.  This UI cut of the rest of the block.